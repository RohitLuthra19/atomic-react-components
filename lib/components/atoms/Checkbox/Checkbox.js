// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import InputControl from '../InputControl';
import styles from './Checkbox.style';
import Label from '../Label/Label';

type Props = {
  id ?: string,
  name: string,
  className ?: string,
  inputClassName ?: string,
  showLabelBeforeInput ?: boolean,
  label ?: string,
  labelProps ?: {},
  children ?: Node
};

const Checkbox = ({
  id,
  className,
  inputClassName,
  showLabelBeforeInput,
  name,
  children,
  label,
  labelProps,
  ...otherProps
}: Props): Node => (
  <Label className={className} htmlFor={id} {...labelProps}>
    {!showLabelBeforeInput && <InputControl id={id} type="checkbox" name={name} className={inputClassName} {...otherProps} />}
    {children || label}
    {showLabelBeforeInput && <InputControl id={id} type="checkbox" name={name} className={inputClassName} {...otherProps} />}
  </Label>
);

Checkbox.defaultProps = {
  id: '',
  className: '',
  inputClassName: '',
  label: '',
  labelProps: {},
  children: '',
  showLabelBeforeInput: false,
};

export default styled(Checkbox)`${styles}`;
