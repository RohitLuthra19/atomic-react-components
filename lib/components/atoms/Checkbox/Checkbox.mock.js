const checkbox = {
  name: 'test_1',
  id: 'test_1',
  value: 'yes',
  label: 'Default Checkbox',
};

const defaultCheckedCheckbox = {
  name: 'test_2',
  id: 'test_2',
  value: 'defaultChecked',
  checked: true,
  label: 'default checked',
};

const disabledCheckbox = {
  name: 'test_3',
  id: 'test_3',
  value: 'disabled',
  label: 'disabled',
  disabled: true,
};

const checkboxWithLabelBeforeInput = {
  name: 'test_4',
  id: 'test_4',
  value: true,
  label: 'Checkbox with label before input',
  showLabelBeforeInput: true,

};

export {
  checkbox,
  defaultCheckedCheckbox,
  disabledCheckbox,
  checkboxWithLabelBeforeInput,
};
