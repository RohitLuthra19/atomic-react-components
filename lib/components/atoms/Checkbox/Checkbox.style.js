import { css } from 'styled-components';

export const inputStyles = css`
  position: absolute;
  opacity: 0;
  label > span {
    &:after {
      content: none;
    }
  }
  &:checked + label > span {
    border-color: ${props => props.theme.brandPrimaryColor};
    background-color: ${props => props.theme.brandPrimaryColor};
    border-radius: 2px;
    &:after {
      content: "";
    }
  }
  ${props => (props.disabled ? css`
    & + label > span {
      border-color: ${props.theme.fieldBorderDisabledColor};
    }
    & + label {
      cursor: default;
    }
  ` : css`
      &:focus + label > span,
      &:hover + label > span {
        border-color: ${props.theme.brandPrimaryColor};
      }
  `)}
`;

export const labelStyles = css`
  cursor: pointer;
  position: relative;
  display: inline-flex;
  align-items: center;
  > span {
    border: 2px solid ${props => props.theme.colorOutline};
    width: 20px;
    height: 20px;
    position: relative;
    margin-right: 5px;

    &::after {
      content: "";
      width: 12px;
      height: 6px;
      border-left: 2px solid;
      border-bottom: 2px solid;
      border-color: #fff;
      left: 3px;
      top: 3px;
      transform: rotate(-45deg);
      position: absolute;
    }
  }
`;
