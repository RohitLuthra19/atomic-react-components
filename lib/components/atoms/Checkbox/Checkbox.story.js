// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  checkbox,
  defaultCheckedCheckbox,
  disabledCheckbox,
  checkboxWithLabelBeforeInput,
} from './Checkbox.mock';
import Checkbox, { CheckboxVanilla } from './Checkbox';

storiesOf('Atoms', module).addWithChapters('Checkbox', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <CheckboxVanilla {...checkbox} id="test" name="test" />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Checkbox Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Checkbox {...checkbox} />,
        },
        {
          title: 'Default Checked',
          sectionFn: () => <Checkbox {...defaultCheckedCheckbox} />,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Checkbox {...disabledCheckbox} />,
        },
        {
          title: 'Checkbox with showLabelBeforeInput prop',
          sectionFn: () => <Checkbox {...checkboxWithLabelBeforeInput} />,
        },
      ],
    },
  ],
});
