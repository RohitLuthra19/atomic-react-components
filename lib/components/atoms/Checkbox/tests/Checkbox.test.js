import React from 'react';
import { shallow } from 'enzyme';
import { CheckboxVanilla } from '../Checkbox';

describe('Checkbox Component', () => {
  let checkbox = '';
  beforeEach(() => {
    checkbox = shallow(<CheckboxVanilla name="test" id="name">Checkbox</CheckboxVanilla>);
  });

  test('should render correctly', () => {
    expect(checkbox).toMatchSnapshot();
  });

  test('should render disabled', () => {
    checkbox = shallow(<CheckboxVanilla name="test" id="name" disabled>Submit</CheckboxVanilla>);
    expect(checkbox.find({ disabled: true })).toHaveLength(1);
  });

  test('should render checked', () => {
    checkbox = shallow(<CheckboxVanilla name="test" id="name" checked>Submit</CheckboxVanilla>);
    expect(checkbox.find({ checked: true })).toHaveLength(1);
  });

  test('should render label as children if label prop is passed', () => {
    checkbox = shallow(<CheckboxVanilla name="test" id="name" label="checkbox label" />);
    expect(checkbox.find('span').text()).toEqual('checkbox label');
  });
});
