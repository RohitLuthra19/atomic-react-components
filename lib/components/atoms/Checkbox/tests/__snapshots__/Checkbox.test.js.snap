// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should render correctly 1`] = `
<Label
  className=""
  htmlFor="name"
>
  <InputControl
    id="name"
    name="test"
    type="checkbox"
  />
  <span>
    Checkbox
  </span>
</Label>
`;
